zero/meta-csp/recipes-cspd/csp-handler/files/camera.c

```diff
@@ -6,11 +6,244 @@
 
 #include <sys/types.h>
 #include <dirent.h>
+#include <linux/videodev2.h>
+#include <sys/stat.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <jpeglib.h>
+#include <unistd.h>
+
 #include "cspd.h"
+#include "yuyv_to_rgb.h"
+
+#define CAMERA_WIDTH  1920
+#define CAMERA_HEIGHT 1080
+#define JPEG_QUALITY 90
+#define MMAP_COUNT    2
 
 #define CAM_FRAME_PATH   "/storageA/photo"
 #define CAM_FRAME_PREFIX "frame"
 
+static int xioctl(int fd, int request, void *arg)
+{
+	for (;;) {
```

Comment:
I don't want to have infinit loops in space grade software.  I know we have it in Zephyr kernel and libcsp right now. But this whole `xioctl()` thing is a way for ordinal software, such as editors, language runtime, and many others.  That's why AI uses it. (I'm assuming it's generated)